modules/directory-bucket/main.tf

```diff
@@ -0,0 +1,144 @@
+locals {
+  bucket_name = "${var.bucket_name_prefix}--${var.availability_zone_id}--x-s3"
+}
+
+resource "aws_s3_directory_bucket" "this" {
```

Comment:
This submodule creates one unique resource type (`aws_s3_directory_bucket`), and the rest are the same as those in the main S3 bucket module.

Given there are [rather big differences in features](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-differences.html), do you think we can have this resource `aws_s3_directory_bucket` in the main module and make the creation of it conditionally (e.g. `is_directory_bucket = true/false`)? This way, we will have less code repetition.

Or am I missing a bigger picture? What do you think?

Reply:
That works as well. I was initially thinking since most of the resources in the main module didn't apply to the directory bucket, it would be easier to have a sub-module, but using `is_directory_bucket` also works. I refactored to this. Let me know if that works.